approvals/approvals/api.py

```diff
@@ -99,16 +107,26 @@ def fetch_approvals_and_roles(doc: Document, method: str | None = None):
 
 @frappe.whitelist()
 def approve_document(
-	doc: Document, method: str | None = None, role: str | None = None, user: str | None = None
+	doc: Document | str, method: str | None = None, role: str | None = None, user: str | None = None
 ):
-	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
+	doc = frappe.get_doc(json.loads(doc)) if isinstance(doc, str) else doc
 	approval = frappe.new_doc("Document Approval")
 	approval.reference_doctype = doc.doctype
 	approval.reference_name = doc.name
 	approval.approver = user
 	approval.approval_role = role if role != "User Approval" else None
 	approval.user_approval = "User Approval" if role == "User Approval" else None
 	approval.save(ignore_permissions=True)
+
+	# TODO: is this required?
+	add_approval_comment(
+		doc.doctype,
+		doc.name,
+		f"Document approved by <b>{frappe.session.user}</b>",
+		subject="Document approved",
+		comment_type="Info",
+	)
```

Comment:
@agritheory I wasn't sure if a comment should be added on successful approval of a document or only when an approver is added/removed, so treat this is a temporary placeholder.